terragrunt/README.md

```diff
@@ -1,20 +1,122 @@
+[Terragrunt] is a wrapper around [Terraform] that adds some additional
+functionality that we find useful. The features of Terragrunt we use are:
 
+  - [Managed state files](https://terragrunt.gruntwork.io/docs/getting-started/quick-start/#keep-your-backend-configuration-dry):
+    Terragrunt can create the S3 bucket and DynamoDB tables that Terraform needs
+    to store its state. Automating their creation ensures consistency across
+    environments.
+  - [Versioned environments](https://terragrunt.gruntwork.io/docs/getting-started/quick-start/#promote-immutable-versioned-terraform-modules-across-environments):
+    With Terragrunt, we can have multiple environments that use different
+    versions of a Terraform module. This enables us to test on staging, review
+    changes, and then update production to the specific version.
 
-## Directory structure
+## Directory Structure
 
-The `terragrunt` directory is split into two subdirectories:
-* `modules`: the reuseable module definitions for the logical services we manage.
-* `aws`: AWS account definitions which use the modules in the `modules` directory to configure the various AWS accounts we manage.
+The [Terragrunt] configuration is split into two parts: _modules_ and _state_:
 
-## Running
+- _Modules_ are normal [Terraform] modules that each provision a specific part
+  of our infrastructure. They are stored in the [`modules`](#modules) directory.
+- _State_ represents concrete instances of our infrastructure. For example, the
+  [staging environment for `crates.io`](./accounts/legacy/crates-io-staging) is
+  an instances of the [`crates-io`](./modules/crates-io) [Terraform] module.
+  State is managed using [Terragrunt], and is configured inside the
+  [`accounts`](#accounts) directory.
 
-Running terragrunt requires permissions to the AWS account you are configuring. Assuming you have permission, you can cd into the corresponding service within the `aws` directory and run `terragrunt plan` to see the plan terraform will apply and `terragrunt apply` to actually apply the plan.
+Both concepts are described in more details below.
+
+### `accounts`
+
+The `accounts` subdirectory contains the _state_ of our infrastructure.
+
+A _state_ is a concrete instance of a _service_ or a shared module, which has
+been defined as a [Terraform] module in [`modules`](#modules). States are
+deployed using [Terragrunt], and are grouped by the account in which they run.
+
+Given the following directory structure:
+
+```text
+accounts
+└── dev-desktops-prod
+    ├── westeurope
+    └── westus2
+```
+
+`dev-desktops-prod` is (AWS) _account_ that stores the Terraform state files for
+this environment. `westeurope` and `westus2` are two _states_, i.e. concrete
+instances of the [`dev-desktops-azure`](./modules/dev-desktops-azure) module.
+
```

Comment:
Perhaps we should clarify why an AWS account houses two instances of Azure based modules - i.e., that the shared terraform state lives in the AWS account.

Reply:
Added a sentence below.